Document that the StyleHandler is final now (#126)
WoltLab/woltlab.github.io: docs/migration/wsc53/php.md

You should take the time to verify the rewritten queries against the manual of the search engine to make sure it cannot generate malformed queries or security issues.

See [WoltLab/WCF#3815](https://github.com/WoltLab/WCF/issues/3815) for details.

## Styles

> The `StyleCompiler` is marked `final` now.

Reviewer: ```suggestion
The `StyleCompiler` class is marked `final` now.
```

> It was noticed that the `$compiler` property was a design issue that leaked compiler state across multiple compiled styles, possibly causing misgenerated stylesheets.

Reviewer: ```suggestion
The internal SCSS compiler object being stored in the `$compiler` property was a design issue that leaked compiler state across multiple compiled styles, possibly causing misgenerated stylesheets.
```

> As the removal of the `$compiler` property effectively broke compatibility within the `StyleCompiler` and as the `StyleCompiler` never was meant to be extended it was marked final.

Reviewer: ```suggestion
As the removal of the `$compiler` property effectively broke compatibility within the `StyleCompiler` and as the `StyleCompiler` never was meant to be extended, it was marked final.
```

> See [WoltLab/WCF#3929](https://github.com/WoltLab/WCF/pull/3929) for details.

Reviewer: ```suggestion
Refer to [WoltLab/WCF#3929](https://github.com/WoltLab/WCF/pull/3929) for details.
```

Author: This one should be consistently adjusted everywhere.
